gensim/models/doc2vec.py

```diff
@@ -794,6 +794,11 @@ def train(self, documents=None, corpus_file=None, total_examples=None, total_wor
 
         """
         kwargs = {}
+
+        # Check the type of corpus_file
+        if not isinstance(corpus_file, string_types):
+            raise TypeError("Parameter corpus_file of train() must be a string (path to a file).")
```

Comment:
Can you also show what the received parameter is instead?

Having concrete error messages helps avoid confusion.